### PR TITLE
Add BT decorator invert

### PR DESCRIPTION
**Pull request by necessarily-equal**
This swaps the STATUS_RUNNING and STATUS_FAILURE states of its child, and otherwise runs it like usual.

```
afontain> freem: I think I would need a decorator that inverts the status code
freem> well, the feature in itself makes sense tbh
freem> can't say I never thought about it :)
```

### CODE REVIEW

**Comment by necessarily-equal** [comment]
apparently due to a parser bug, you must use

```
decorator invert() {
    …
}
```

instead of

```
decorator invert {
    …
}
```

although it's not a big deal, both are supposed to be accepted in the source code.

File: src/sgame/sg_bot_ai.cpp [hunk 1]
```diff
@@ -325,6 +325,22 @@ Decorators
...
 ======================
 */
+
+AINodeStatus_t BotDecoratorInvert( gentity_t *self, AIGenericNode_t *node )
+{
+	AIDecoratorNode_t *dec = ( AIDecoratorNode_t * ) node;
+
+	AINodeStatus_t status = BotEvaluateNode( self, dec->child );
+
+	if ( status == STATUS_SUCCESS )
```
Comment (bmorel) [inline src/sgame/sg_bot_ai.cpp:335]: a switch seems to be what you really want here.

File: src/sgame/sg_bot_ai.cpp [hunk 2]
```diff
@@ -325,6 +325,22 @@ Decorators
 Decorators are used to add functionality to the child node
 ======================
 */
+
+AINodeStatus_t BotDecoratorInvert( gentity_t *self, AIGenericNode_t *node )
+{
+	AIDecoratorNode_t *dec = ( AIDecoratorNode_t * ) node;
```
[the remark this answers is not in the document]
Comment (necessarily-equal) [reply src/sgame/sg_bot_ai.cpp:331]: I'm following the convention used in other BotDecorator*. IMO this is one of the cases where it's better to modernise every instance at once, or not at all.

File: src/sgame/sg_bot_ai.cpp (hunk 1 above)
Comment (necessarily-equal) [reply src/sgame/sg_bot_ai.cpp:335]: maybe ? idk I like it like this

File: src/sgame/sg_bot_ai.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmorel) [reply src/sgame/sg_bot_ai.cpp:331]: Well, in my changes I'm trying to move away from C-style casts as much as possible. I guess I'll just have to do it on one more thing. This is not blocking.

**Review by bmorel — APPROVED** [review]
LGTM

**Pull request merged**